plugins/CoreHome/vue/src/NumberFormatter/NumberFormatter.ts

```diff
@@ -0,0 +1,256 @@
+          return true;
+        }
+
+        return false;
+      });
+
+      newValue += valueReplaced;
+    });
+
+    return newValue;
+  }
+
+  private valOrDefault(def: number, val?: number): number {
+    if (typeof val === 'undefined') {
+      return def;
+    }
+
+    return val;
+  }
+
+  public formatNumber(
+    value: string|number,
+    maxFractionDigits?: number,
+    minFractionDigits?: number,
+  ): string {
+    return this.format(
+      value,
+      Matomo.numbers.patternNumber,
+      this.valOrDefault(this.defaultMaxFractionDigits, maxFractionDigits),
+      this.valOrDefault(this.defaultMinFractionDigits, minFractionDigits),
+    );
+  }
+
+  public formatPercent(
+    value: string|number,
+    maxFractionDigits?: number,
+    minFractionDigits?: number,
+  ): string {
+    return this.format(
+      value,
+      Matomo.numbers.patternPercent,
+      this.valOrDefault(this.defaultMaxFractionDigits, maxFractionDigits),
+      this.valOrDefault(this.defaultMinFractionDigits, minFractionDigits),
+    );
+  }
+
+  public formatCurrency(
+    value: string|number,
+    currency: string,
+    maxFractionDigits?: number,
+    minFractionDigits?: number,
+  ): string {
+    const formatted = this.format(
+      value,
+      Matomo.numbers.patternCurrency,
+      this.valOrDefault(this.defaultMaxFractionDigits, maxFractionDigits),
+      this.valOrDefault(this.defaultMinFractionDigits, minFractionDigits),
+    );
+    return formatted.replace('¤', currency);
+  }
+
+  public formatEvolution(
+    evolution: string|number,
+    maxFractionDigits?: number,
+    minFractionDigits?: number,
+    noSign?: boolean,
+  ): string {
+    if (noSign) {
+      return this.formatPercent(
+        Math.abs(evolution as number),
+        maxFractionDigits,
+        minFractionDigits,
+      );
+    }
+    const formattedEvolution = this.formatPercent(evolution, maxFractionDigits, minFractionDigits);
+    return `${evolution as number > 0 ? Matomo.numbers.symbolPlus : ''}${formattedEvolution}`;
+  }
+
+  public calculateAndFormatEvolution(
+    currentValue: string|number,
+    pastValue: string|number,
+    noSign?: boolean,
+  ) {
+    const pastValueParsed = parseInt(pastValue as string, 10);
+    const currentValueParsed = parseInt(currentValue as string, 10) - pastValueParsed;
+
+    let evolution: number;
+
+    if (currentValueParsed === 0 || Number.isNaN(currentValueParsed)) {
+      evolution = 0;
+    } else if (pastValueParsed === 0 || Number.isNaN(pastValueParsed)) {
+      evolution = 100;
+    } else {
+      evolution = (currentValueParsed / pastValueParsed) * 100;
+    }
+
+    let maxFractionDigits = 3;
+
+    if (Math.abs(evolution) > 100) {
+      maxFractionDigits = 0;
+    } else if (Math.abs(evolution) > 10) {
+      maxFractionDigits = 1;
+    } else if (Math.abs(evolution) > 1) {
+      maxFractionDigits = 2;
+    }
+
+    return this.formatEvolution(evolution, maxFractionDigits, 0, noSign);
```

Comment:
The title tooltip for the old/current All Websites Dashboard goes wild:

> 666 visits in 2024-02-08 compared to 10,309 in 2024-01-08. Evolution: +6,560%

Previously it was `-94%` :see_no_evil: 

It is now broken "slightly more than before", due to the formatting applied to the past data in the API response. The `,` in my past data response is a problem, as it will be parsed from `10,309` to `10`, while previously the evolution worked if the current value was below 1k (no matter the past value). Now it is broken if either of the values is above 1k. 

Maybe we need to check for `symbolMinus`, split on `symbolDecimal`, remove any non-digit numbers, and then recombine to a number type to get strings to a more usable state?